components/engine_traits/src/engine.rs

```diff
@@ -20,30 +21,77 @@ pub trait WriteBatch: Mutable {
     fn rollback_to_save_point(&mut self) -> Result<()>;
 }
 
-pub trait KvEngine: Peekable + Mutable + Iterable + Send + Sync + Clone {
+pub trait KvEngine: Peekable + Mutable + Iterable + Send + Sync + Clone + Debug + 'static {
     type Snap: Snapshot;
     type Batch: WriteBatch;
+    type DBOptions: DBOptions;
```

Comment:
I'd like to keep the core trait `KvEngine` as simple as possible. New associated types can be added to a new trait, like `trait KvEngineExt`, which can be bounded by `KvEngine`.

Reply:
Ok, I'll move these to another trait.

Reply:
Maybe instead of a catch-all KvEngineExt trait I will go ahead and make traits for specific purposes.

Reply:
This is done.